source/cli.py

```
@@ -85,12 +91,23 @@ def make_missed_class_table(response, attendance, cached_yes):
         result.append([data['code'], data['Date'], data['subjectName'], data['attended'] + '/' + data['total']]) 
 
     today = datetime.date.today()
+    file = os.path.expanduser("~/.attendance_past.txt")
+
+    try:
+        win32api.SetFileAttributes(file, win32con.FILE_ATTRIBUTE_NORMAL)
+    except Exception:
+        pass
```

If a hidden file can't be opened, then upon failing here will lead to another error which you have not written in try-catch style.

---

If you note, I set the file attributes to Normal (if possible) before doing any operations on it [here](https://github.com/dsciitbbs/acl/pull/9/files#diff-86fec04b8cb29aeaf74e237511ec2dd0R96-R98)

So, I am unsure what possible exception could be raised. If the file cannot be saved/opened, the exception would be swallowed by the block, right?

---

Do you mean the possibility that the file attribute setting will fail but file open call will work? To guard against that possibility, we can bring it inside the try/except block, but to be honest, it's going to be very rare and then we should wrap all file I/O blocks inside ```try/except``` to be **safe**, which is just weird.

I could move it if needed, but is it necessary? 🤔

---

If line 97 fails, for some reason, line 101 will fail and this will lead to application crash.